text/logs/0150-logging-library-sdk.md

```diff
@@ -0,0 +1,255 @@
+# Logging Library SDK Prototype Specification
+
+Initial draft specification to create prototypes of OpenTelemetry Logging
+Library SDKs.
+
+Status: for prototyping. Don't merge to main OpenTelemetry specification until
+prototyping is done.
+
+## Motivation
+
+This is a draft proposal for OpenTelemetry Logging Library SDK specification.
+The purpose of his proposal is to lay out the initial understanding of
+what Logging Library SDKs will look like.
+
+We would like to create prototypes based on this proposal in a few languages.
+The learnings from the prototypes will then allow us to refine this proposal and
+eventually make the proposed approach part of the OpenTelemetry specification.
+
+The approach proposed in this OTEP is not intended to be merged into the
+OpenTelemetry specification after the OTEP itself is approved. We want to have
+successful prototypes first. This OTEP is first and foremost a specification and
+guidelines on how to create the prototypes.
+
+The specification defines how the OpenTelemetry Logging Library SDK exposes its
+functionality to authors of extensions to language-specific 3rd party logging
+libraries and to end users that want to produce logs in the
+[OpenTelemetry manner](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/overview.md).
+
+The specification defines SDK elements that to some extent mirror the
+OpenTelemetry
+[Trace SDK](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace).
+This ensures uniformity and consistency of the OpenTelemetry specification and
+of the implementations across traces and logs. For additional clarity the
+definitions in this document refer to the Trace analogs where appropriate.
+
+The descriptions of interfaces and methods in this document are intentionally
+very brief. Detailed and precise descriptions will be borrowed from Trace
+specification when this OTEP is submitted as a Log specification PR. I kept the
+descriptions as short as possible in this document to make reviewing it easy and
+fast.
+
+## Specification
+
+Many existing logging libraries have some sort of extension mechanism that
+allows to customize how log records are encoded and delivered to their
+destinations (for example Appender in Log4j or Core in zapcore). The
+OpenTelemetry Logging Library SDK is intended to be used by such extensions to
+emit logs in OpenTelemetry formats.
+
+Note: The functionality that this document proposes will be an SDK package. A
+logging-related API package may be added in the future if we decide to have an
+end-user callable logging API.
+
+We will have the following in the SDK.
+
+### LogEmitterProvider
+
+Methods:
+
+- Get LogEmitter. Accepts the instrumentation library name and version and
```

Comment:
As part of current cpp [implementation ](https://github.com/open-telemetry/opentelemetry-cpp/blob/1e5751e20367b715221bf3820b84590559170cdf/sdk/include/opentelemetry/sdk/logs/logger_provider.h#L60), we are using logger-name instead of instrumentation library name to get the correct `Logger`. I understand there have been some discussion on including the logger-name in log-record data model, not sure the current status though.

Reply:
Yes, https://github.com/open-telemetry/opentelemetry-specification/pull/1236 suggests that the instrumentation library name and logger name are distinct and both needed.